### ISSUE TITLE
Circular header dependency

### ISSUE THREAD

**Issue by Bolderaysky** [stephenberry/glaze#846]
After updating my codebase to latest version of glaze, i found that it no longer compiles. After a bit of research, I saw the following headers `#include` chain that causes this issue:
```
glaze/reflection/get_name.hpp -> glaze/api/name.hpp -> glaze/core/meta.hpp -> glaze/reflection/get_name.hpp
```

After commenting out `#include glaze/api/name.hpp` in `glaze/reflection/get_name.hpp`, both my code and glaze tests successfully compiles, so it looks like this doesn't have any side effect.

Interesting, even considering this circular dependency, glaze tests and actions continue to compile and it would be non-trivial task  to make reproducible example here. Anyway, circular dependencies can cause unexpected problems, so it's best to handle that

**Comment by stephenberry** [comment]
Thanks for pointing this out. Will fix.

**Comment by stephenberry** [comment]
This should be fixed as of #849
